language/move-core/types/src/language_storage.rs

```diff
@@ -79,6 +103,153 @@ impl<'a> From<&'a TypeTag> for SafeTypeTag<'a> {
     }
 }
 
+// u8 -> depth or nesting level
+struct TypeTagVisitor(u8);
+
+use serde::de::{
```

Comment:
Could you please ensure to format before sending? I was spending a while wondering where your Visitor and other types come from, as I did not see them imported on the top of the file (where expected).